### Simple Regression Problem solved using dense layers in a DNN design
Uses 5-fold cross-validation with sklearn pipeline

In [15]:
import numpy
import pandas
from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras.wrappers.scikit_learn import KerasRegressor
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline

In [2]:
# load dataset
dataframe = pandas.read_csv("housing.data", delim_whitespace=True, header=None)
dataset = dataframe.values
# split into input (X) and output (Y) variables
X = dataset[:,0:13]
Y = dataset[:,13]

In [16]:
# define base model
def baseline_model():
    # create model
    model = Sequential()
    model.add(Dense(128, input_dim=13, kernel_initializer='normal', activation='relu'))
    model.add(Dropout(0.25))
    model.add(Dense(1, kernel_initializer='normal'))
    # Compile model
    model.compile(loss='mean_squared_error', optimizer='adam',metrics=['accuracy'])
    return model

In [23]:
# evaluate model with standardized dataset
seed = 1
numpy.random.seed(seed)
estimators = []
estimators.append(('standardize', StandardScaler()))
estimators.append(('mlp', KerasRegressor(build_fn=baseline_model, epochs=100, batch_size=25, verbose=0)))
pipeline = Pipeline(estimators)
kfold = KFold(n_splits=5, random_state=seed)
results = cross_val_score(pipeline, X, Y, cv=kfold)

In [24]:
print("Standardized: %.2f (%.2f) MSE" % (results.mean(), results.std()))

Standardized: 26.72 (16.21) MSE


### Results
Standardized: 41.81 (18.85) MSE
Standardized: 43.79 (17.27) MSE - Dropout
Standardized: 39.53 (17.15) MSE - Dropout, 30 epochs
Standardized: 37.07 (15.51) MSE - Dropout, 40 epochs
Standardized: 26.72 (16.21) MSE - Dropout, 100 epochs